pkg/controller/certificates/util_test.go

```diff
@@ -302,7 +302,7 @@ func TestRenewalTime(t *testing.T) {
 		renewBeforeOverride *metav1.Duration
 		expectedRenewalTime *metav1.Time
 	}
-	now := time.Now()
+	now := time.Now().Truncate(time.Second)
```

Comment:
Thanks, that is a really good point, this should be tested better! I've added a test case which is the same as what I saw with LetsEncrypt where the second value for 'notAfter' is 'notBefore' - 1. Before this fix this would have caused the returned value to be `<truncated-value-to-seconds>.333333333`

I think in most cases this bug will happen when the cert's `notAfter` - `notBefore` does not divide by 3 (usually it will since it's normally 24h, 60m etc) i.e because there is the 1 second skew. I don't know if it ever happens that the `notBefore`, `notAfter` values themselves may have non-zero values for millisecond etc units.
Let me know if you think this test is sufficient.

Reply:
The new test looks good! I think I'd still prefer to see some tests with hard-coded times rather than just using `Now()` because `Now()` isn't repeatable, by definition. But that's probably a change outside of the scope of this PR, and I think the new test is good.